proposals/gc/MVP.md

```diff
@@ -604,33 +604,36 @@ In particular, `ref.null` is typed as before, despite the introduction of `none`
 
 Casts work for both abstract and concrete types. In the latter case, they test if the operand's RTT is a sub-RTT of the target type.
 
-* `ref.test <reftype>` tests whether a reference has a given type
-  - `ref.test rt : [rt'] -> [i32]`
-    - iff `rt <: trt` and `rt' <: trt` for some `trt`
-  - if `rt` contains `null`, returns 1 for null, otherwise 0
-
-* `ref.cast <reftype>` tries to convert a reference to a given type
-  - `ref.cast rt : [rt'] -> [rt]`
-    - iff `rt <: trt` and `rt' <: trt` for some `trt`
+* `ref.test null? <heaptype>` checks whether a reference has a given heap type
+  - `ref.test null? ht : [(ref null ht')] -> [i32]`
+    - iff `ht <: tht` and `ht' <: tht` where `tht` is a common super type
+  - if `null?` is present, returns 1 for null, otherwise 0
+
+* `ref.cast null? <heaptype>` tries to convert to a given heap type
+  - `ref.cast null? ht : [(ref null ht')] -> [(ref null2? ht)]`
+    - iff `ht <: tht` and `ht' <: tht` where `tht` is a common super type
+    - and `null? = null2?`
```

Comment:
Can we just use `null?` again in the typing rule instead of introducing `null2?`?

Reply:
done